pandas/tests/reshape/test_pivot.py

```diff
@@ -2058,6 +2058,81 @@ def test_pivot_string_as_func(self):
         ).rename_axis("A")
         tm.assert_frame_equal(result, expected)
 
+    @pytest.mark.parametrize("kwargs", [{"a": 2}, {"a": 2, "b": 3}, {"b": 3, "a": 2}])
+    def test_pivot_table_kwargs(self, kwargs):
+        def f(x, a, b=3):
+            return x.sum() * a + b
+
+        def g(x):
+            return f(x, **kwargs)
+
+        df = DataFrame(
+            {
+                "A": ["good", "bad", "good", "bad", "good"],
+                "B": ["one", "two", "one", "three", "two"],
+                "X": [2, 5, 4, 20, 10],
+            }
+        )
+        result = pivot_table(
+            df, index="A", columns="B", values="X", aggfunc=f, **kwargs
+        )
+        expected = pivot_table(df, index="A", columns="B", values="X", aggfunc=g)
+        tm.assert_frame_equal(result, expected)
+
+        expected = DataFrame(
+            [[np.nan, 43.0, 13.0], [15.0, np.nan, 23.0]],
+            columns=Index(["one", "three", "two"], name="B"),
+            index=Index(["bad", "good"], name="A"),
+        )
+        tm.assert_frame_equal(result, expected)
+
+    @pytest.mark.parametrize(
+        "kwargs", [{}, {"b": 10}, {"a": 3}, {"a": 3, "b": 10}, {"b": 10, "a": 3}]
+    )
+    def test_pivot_table_kwargs_margin(self, data, kwargs):
+        def f(x, a=5, b=7):
+            return (x.sum() + b) * a
+
+        def g(x):
+            return f(x, **kwargs)
+
+        result = data.pivot_table(
+            values="D",
+            index=["A", "B"],
+            columns="C",
+            aggfunc=f,
+            margins=True,
+            fill_value=0,
+            **kwargs,
+        )
+
+        expected = data.pivot_table(
+            values="D",
+            index=["A", "B"],
+            columns="C",
+            aggfunc=g,
+            margins=True,
+            fill_value=0,
+        )
+
+        tm.assert_frame_equal(result, expected)
+
+        grand_margin = g(data["D"])
+
+        margin_col = pivot_table(
+            data, values="D", index=["A", "B"], aggfunc=g, fill_value=0
+        )
+        margin_col.loc[("All", ""), "D"] = grand_margin
+        margin_col = margin_col["D"]
+        margin_col.name = "All"
+        tm.assert_series_equal(result["All"], margin_col)
+
+        margin_row = pivot_table(data, values="D", columns="C", aggfunc=g, fill_value=0)
+        margin_row["All"] = grand_margin
+        margin_row.index = [""]
+        margin_row.index.name = "B"
+        tm.assert_frame_equal(result.loc["All"], margin_row)
```

Comment:
Similar remarks here - the above tests all of `result`, why do we need to test `result.loc["All"]`?